files/shared_task_functions.ps1

```diff
@@ -0,0 +1,303 @@
+function Write-BoltError {
+  param(
+    $message,
+    $errorObject
+  )
+
+  $errorReturn = @{
+    _error = @{
+      msg     = $message
+      kind    = 'puppetlabs.task/task-error'
+      details = @{
+        detailedInfo = $errorObject.exception.message
+        exitcode     = 1
+      }
+    }
+  }
+
+  $errorReturn | ConvertTo-JSON -depth 99
+}
+
+function Select-LoginName {
+  param(
+    [PSObject]$login,
+    [string[]]$namesToMatch,
+    [switch]$exact_match
+  )
+
+  # This function takes a single SQLServer login object and compares it against
+  # the list of names passed into the -login_name parameter of the script to
+  # determine if this is a login the user is interested in seeing. If it does
+  # not pass the filter represented by that parameter the login is discarded.
+
+  foreach ($paramLogin in $namesToMatch) {
+    if ($exact_match) {
+      if ($paramLogin -eq $login.name) {
+        Write-Output $login
```

Comment:
I know it was previously my own advice that made `Write-Output` show up here, but I've since changed my tune due to some community info and practices:

> Avoid Write-Output. Instead, when you want to make output clearer, just assign output to a relevantly named variable. and put that a variable on a line by itself to signal explicit output. In particular, avoid `Write-Output -NoEnumerate` which is [broken in PowerShell 6](https://github.com/PowerShell/PowerShell/issues/5955).
>
> https://github.com/PoshCode/PowerShellPracticeAndStyle/issues/46#issuecomment-236727676

Reply:
I don't think I agree that putting a variable on a line by itself is clearer, but `(and worse, can be overwritten by functions or aliases)`, does at least seem like a thing that could actually happen, I'll go ahead and make the switch.